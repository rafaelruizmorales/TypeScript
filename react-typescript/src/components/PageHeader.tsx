type PageHeaderProps = {
    title: string
}

export const PageHeader: React.FC<PageHeaderProps> = ({title}) => {
    return (
        <div className="page-header">
            <h1>{title}</h1>
        </div>
    )
}