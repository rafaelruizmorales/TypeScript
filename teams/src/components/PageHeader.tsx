import * as React from 'react'

interface OwnProps {
  title: string
}

export const PageHeader: React.FC<OwnProps> = ({title}) => {
  return <div className="mb-3 mt-3"><h1>{title}</h1></div>
}
