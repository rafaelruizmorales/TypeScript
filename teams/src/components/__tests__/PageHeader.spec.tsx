import {shallow} from 'enzyme'
import * as React from 'react'
import {PageHeader} from '../PageHeader'

describe('PageHeader component', () => {
  it('should match the snapshot', () => {
    expect(shallow(<PageHeader title="Page Header" />)).toMatchSnapshot()
  });

  it('should render the title inside an H1 tag', () => {
    const title = 'Page Header'
    const component = shallow(<PageHeader title={title} />)
    expect(component.find('h1').text()).toBe(title)
  });
});
