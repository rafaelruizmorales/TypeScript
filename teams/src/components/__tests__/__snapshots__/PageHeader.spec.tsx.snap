// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component should match the snapshot 1`] = `
<div
  className="mb-3 mt-3"
>
  <h1>
    Page Header
  </h1>
</div>
`;
